base/special/gamma.jl

```diff
@@ -482,3 +482,7 @@ eta(x::Integer) = eta(Float64(x))
 eta(x::Real)    = oftype(float(x),eta(Float64(x)))
 eta(z::Complex) = oftype(float(z),eta(Complex128(z)))
 @vectorize_1arg Number eta
+
+Base.binomial{S<:Number,T<:Number}(n::S, k::T) = ( (isinteger(n) && isinteger(k))
```

Comment:
The general practice in Julia is to use native `Int` by default, to avoid imposing an overhead on all use cases just to support very specific cases. Always using `BigInt` goes against this idea.